pkg/utils/environment_variable.go

```diff
@@ -0,0 +1,68 @@
+package utils
+
+import (
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/cyberark/conjur-authn-k8s-client/pkg/log"
+)
+
+// The function 'envVarValueOrDefault' is defined below. We set this var at the
+// top of the file so when we use the function it is clear that it's the default one.
+var defaultEnvVarValueOrDefaultFunc = envVarValueOrDefault
+
+type EnvVarValueOrDefaultFunc func(envVarName string, defaultValue string) string
+
+// IntFromEnvOrDefault extracts the value of the given environment variable,
+// or the default value if the environment variable is not defined, and returns
+// it as an int object
+func IntFromEnvOrDefault(
+	envVarName string,
+	defaultValue string,
+	envVarValueOrDefaultFunc EnvVarValueOrDefaultFunc,
+) (int, error) {
+	if envVarValueOrDefaultFunc == nil {
+		envVarValueOrDefaultFunc = defaultEnvVarValueOrDefaultFunc
+	}
+
+	envVarValue := envVarValueOrDefault(envVarName, defaultValue)
+	envVarValueInt, err := strconv.Atoi(envVarValue)
+	if err != nil {
+		// We return -1 as we can't return nil for int. This value will not be used
+		// as err is not nil.
+		return -1, log.RecordedError(log.CAKC010E, envVarName, err.Error())
```

Comment:
nice, great to learn this!

@sigalsax, @izgeri  ^

Reply:
So I am understanding correctly if we return an error (for example `return -1, log.RecordedError(log.CAKC010E, envVarName, err.Error())`), no matter what, Golang will zero the return values? So for example, -1 will be 0 because it is an integer?

Reply:
i haven't verified it but in any case we will return 0 to maintain the standard